schema/elements_test.go

```diff
@@ -50,6 +50,43 @@ func TestFindNamedType(t *testing.T) {
 	}
 }
 
+func strptr(s string) *string { return &s }
+
+func TestFindField(t *testing.T) {
+	tests := []struct {
+		testName          string
+		defs              []StructField
+		fieldName         string
+		expectStructField StructField
+		expectExist       bool
+	}{
+		{"existing", []StructField{
+			{Name: "a", Type: TypeRef{NamedType: strptr("a")}},
+			{Name: "b", Type: TypeRef{NamedType: strptr("b")}},
+		}, "a", StructField{Name: "a", Type: TypeRef{NamedType: strptr("a")}}, true},
+		{"notExisting", []StructField{
+			{Name: "a", Type: TypeRef{NamedType: strptr("a")}},
+			{Name: "b", Type: TypeRef{NamedType: strptr("b")}},
+		}, "c", StructField{}, false},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.testName, func(t *testing.T) {
+			t.Parallel()
+			s := Map{
+				Fields: tt.defs,
+			}
+			sf, exist := s.FindField(tt.fieldName)
+			if !reflect.DeepEqual(sf, tt.expectStructField) {
```

Comment:
Side note: this usage of DeepEqual illustrates why this "embedded private cache field" isn't a good pattern...

Not asking for a change just noting.

Reply:
Wait, that's not the right code

Reply:
Yeah, we do have an equal function, I could have used that.

Reply:
Done!